SecretSanta/src/SecretSanta.Business/GroupRepository.cs

```diff
@@ -45,5 +47,53 @@ public void Save(Group item)
 
             MockData.Groups[item.Id] = item;
         }
+
+        public AssignmentResult GenerateAssignments(int id)
+        {
+            //grab and do some checks
+            Group group = GetItem(id);
+
+            if (group is null)
+            {
+                throw new NullReferenceException();
+            }
+            if (group.Users.Count < 3)
+            {
+                return AssignmentResult.Error("Sorry, there must be a minimum of three members in a group.");
+            }
+            //begin randomizing.
+            group.Assignments.Clear();
+
+            List<User> randomizedReceiverList = group.Users;
+            Shuffle(randomizedReceiverList);
+            
+            for (int i = 0; i < group.Users.Count; i++)
+            {
+                MockData.Groups[id].Assignments.Add(i < randomizedReceiverList.Count - 1
+                    ? new Assignment(randomizedReceiverList[i], randomizedReceiverList[i + 1])
+                    : new Assignment(randomizedReceiverList[i], randomizedReceiverList[0]));
+            }
+            return AssignmentResult.Success();
+        }
+
+        //list shuffling method from stack overflow
+        //source - https://stackoverflow.com/questions/273313/randomize-a-listt
+        private void Shuffle<T>(IList<T> list)
+        {
+            if (list == null)
+            {
+                throw new NullReferenceException();
```

Comment:
In this case the correct exception to throw would be one of the argument exceptions since this is passed as an argument.

```suggestion
                throw new ArgumentNullException(nameof(list));
```